R/reindent.R

```diff
@@ -0,0 +1,134 @@
+
+#' Update the indention reference
+#'
+#' @param pd_nested A nested parse table.
+#' @name update_indention_ref
+NULL
+
+
+#' @describeIn update_indention_ref Updates the reference id for all
+#'   tokens in `pd_nested` if `pd_nested` contains a function call. Tokens that
+#'   start on the same line as the opening parenthesis, are not themselves
+#'   function calls or expressions wrapped in curly brackets are re-indented,
+#'   that is, they are indented up to the level at which the call ends in
+#'   terms of col2. We need to take the last from the first child because calls
+#'   like package::function() can have three elements.
+#' @examples
+#' \dontrun{
+#' # not re-indented
+#' call(call(
+#'   xyz
+#' ))
+#' # re-indented
+#' call(call(1,
+#'           2))
+#' }
+#' @importFrom purrr map_lgl
+update_indention_ref_fun_call <- function(pd_nested) {
+  current_is_call <- pd_nested$token_before[2] %in% c("SYMBOL_FUNCTION_CALL")
+  non_comment <- which(pd_nested$token != "COMMENT")
+  first_non_comment_after_call <- non_comment[non_comment > 2][1]
+  if ((current_is_call) &&
+      nrow(pd_nested) > 3 &&
+      pd_nested$lag_newlines[first_non_comment_after_call] == 0) {
+    candidates <- 3:(nrow(pd_nested) - 1)
+
+    child_is_call <- map_lgl(pd_nested$child, is_function_call)
+    child_is_curly_expr <- map_lgl(pd_nested$child, is_curly_expr)
+    child_is_on_same_line <- cumsum(pd_nested$lag_newlines) == 0
+    call_on_same_line <- child_is_call & child_is_on_same_line
+    to_indent <- setdiff(candidates, which(call_on_same_line | child_is_curly_expr))
+
+    pd_nested$indention_ref_id[to_indent] <- last(pd_nested$child[[1]]$id)
+  }
+  pd_nested
+}
+
+#' @describeIn update_indention_ref Updates the reference id for all
+#'   tokens in `pd_nested` if `pd_nested` contains a function declaration.
+#'   Tokens inside a function declaration are are re-indented,
+#'   that is, they are indented up to the level at which the token FUNCTION
+#'   ends in terms of col2.
+#' @examples
+#' \dontrun{
+#' a <- function(x,
+#'               y) {
+#' x + y
+#' }
+#' }
+update_indention_ref_fun_dec <- function(pd_nested) {
+  if (pd_nested$token[1] == c("FUNCTION") &&
+      nrow(pd_nested) > 3) {
+    seq <- 3:(nrow(pd_nested) - 1)
```

Comment:
Maybe add a safe `seq_from_to()` helper that will return an increasing sequence starting at `from` and ending at `to - 1`, empty if `from >= to`? I really think `:` should be avoided in package code; you do check for safety but this adds complexity to the code (here and elsewhere).